### PR TITLE
New Uplink Item: Dirty Bomb Remote (Radstorm Remote)

### PR DESCRIPTION
**Pull request by Eneocho**
### Wait no more, pardner! Introducin' the Dirty Bomb Remote!
image
_An agent sending the signal to fire the artillery piece_

Upon pressing the button, a dirty bomb will be teleported to a direct intercept with the station and immediately detonated, spreading radioactive particulate identical to a radiation storm. 
The bluespace artillery piece takes 15 minutes to reload, and the cannon will not fire if a radiation storm is ongoing, either natural or artificial (For various reasons, the radiation storm is considered ongoing up to the moment the maint access gets revoked).
A resourceful agent can take advantage of the precious seconds of radstorm to do whatever it is they require, assuming they have proper radiation protection, or a can do attitude. 
Or maybe the crew is on its last legs and you wish to kick them while they're down? 
Perhaps you sabotage all maintenance airlocks then send a radstorm? So many possibilities!
## But surely this device must be terribly expensive!
Fear not! Thanks to electronics blueprints stolen from NanoTrasen's Bleeding Edge laboratories, this device only costs **12 TC** to acquire! (Or **10 if Engineering**).
## There's got to be a catch!
Sadly there is... Due to NanoTrasen's constant monitoring around their stations, your approximate location will be sent on a report straight to all Communications Consoles (in the same way as the ion cannon gets reported), but maybe you can use this to frame someone else!
<br>
#### Also makes the radstorm event actually have an end time instead of just running for 5 seconds, but that's backend stuff.
:cl:
 * rscadd: Traitors can now buy the Dirty Bomb Artillery Remote, a device that detonates a dirty bomb in orbit and causes a radstorm on-demand. This storm behaves as a regular radstorm, but will report the area the remote was used on upon notifying the crew.

### CODE REVIEW

File: code/game/objects/items/devices/radstorm_remote.dm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	desc = "Press the button for free powers!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "rad_remote"
+	w_class = W_CLASS_TINY
+	flags = FPRINT
+	var/cooldown = 0
+	mech_flags = MECH_SCAN_FAIL
+
+/obj/item/device/radstorm_remote/New()
+	processing_objects.Add(src)
```
Comment (west3436) [inline code/game/objects/items/devices/radstorm_remote.dm:12]: Why are you adding this to processing_objects just to update the icon?

Comment (west3436) [reply code/game/objects/items/devices/radstorm_remote.dm:12]: Disregard, I see you're just mirroring the LOIC implementation

File: code/game/objects/items/devices/radstorm_remote.dm [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/obj/item/device/radstorm_remote
+	name = "\improper suspicious radio"
```
Comment (Kurfursten) [inline code/game/objects/items/devices/radstorm_remote.dm:2]: improper isn't necessary because this isn't capitalized

**Comment by Kurfursten** [comment]
I looked this over and it looks mostly good. You tested?

**Comment by Eneocho** [comment]
Let me see if I can change the device icon to the more menacing once before merging, but yes this is tested extensively while making it since it wouldn't work properly (the event was broken and all). Also, most of the code is straight up lifted from the LOIC.

**Comment by Eneocho** [comment]
Couldn't find the icon and I don't feel like dealing with github's half deleted history to see if I can maybe find it. It's merge ready @Kurfursten.

**Comment by Eneocho** [comment]
actually, i kinda wanna add a 5% chance for the NUCLEAR LAUNCH DETECTED audio to play, since i doubt it'll be noticed during april fools

**Comment by Eneocho** [comment]
updating wiki

**Pull request merged**